### PR TITLE
Feature/cast dates

### LINKED ISSUE #78: No column are displayed when non numeric column is converted to number.

### LINKED ISSUE #79: Support casting columns to date and datetime

### PR DESCRIPTION
**Pull request by bewakes**
Resolves #78 
Resolves #79 

## Changes

* Fix issue when string is wrongly cast to number
* Add casts to date and datetime

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [x] translations

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,6 +24,8 @@ django-redis = "==5.0.0"
 pandas = "*"
 openpyxl = "*"
 jsonschema = "^4.17.3"
+dateparser = "^1.1.7"
+xlrd = "^2.0.1"
```
Comment (k9845) [inline pyproject.toml:28]: Why is this required?

Comment (bewakes) [reply pyproject.toml:28]: Well, for one of the xl files, pandas complained because xlrd was missing.

Comment (k9845) [reply pyproject.toml:28]: xlrd is used as engine in pandas while reading old excel file..didn't new update in pandas helped?

Comment (bewakes) [reply pyproject.toml]: pandas is the latest. Maybe it was an issue with the file. Well, I guess we can get rid of xlrd for the moment and see if the real users get any issues related to xlrd.

**Review by k9845 — APPROVED** [review]
